crates/turborepo-lib/src/run/mod.rs

```diff
@@ -0,0 +1,102 @@
+mod context;
+mod graph;
+pub mod pipeline;
+mod scope;
+mod task_id;
+
+use anyhow::{Context as ErrorContext, Result};
+use graph::CompleteGraph;
+use tracing::{debug, info};
+
+use crate::{
+    commands::CommandBase,
+    daemon::DaemonConnector,
+    manager::Manager,
+    opts::Opts,
+    package_json::PackageJson,
+    run::{context::Context, task_id::ROOT_PKG_NAME},
+};
+
+#[derive(Debug)]
+pub struct Run<'a> {
+    base: &'a CommandBase,
+    opts: Opts<'a>,
+    processes: Manager,
+}
+
+impl<'a> Run<'a> {
+    pub fn new(base: &'a CommandBase, opts: Opts<'a>) -> Self {
```

Comment:
Would it make more sense for the `Run` to own the `CommandBase`? Do we continue using it after executing a command?

Reply:
I've thought about having all the commands be methods of CommandBase, that way we can have all of the relevant resources be owned by CommandBase